lib/Dialect/Handshake/HandshakeOps.cpp

```diff
@@ -1445,6 +1445,63 @@ LogicalResult InstanceOp::verify() {
   return success();
 }
 
+ParseResult UnpackOp::parse(OpAsmParser &parser, OperationState &result) {
+  OpAsmParser::UnresolvedOperand tuple;
+  TupleType type;
```

Comment:
Related to general comment

Reply:
I don't understand what you mean with that. There is already a `TypesMatchWith` trait that verifies the equality of the element types with the types encapsulated in the `TupleType`. 

I could change the builder to accept an additional `TupleType` or a `TypeRange` to infer the types of the results. A type mismatch would later on trigger a verification error, but the builder would run through.

Reply:
You're right, sorry - i missed that `TypesMatchWith`!

Reply:
No worries;)
What about the builder issues? Should I go with the assertion or change it to accept a `TypeRange`?

Reply:
I think the current implementation is fine (single value as input + assert); use of the builder is internal in the compiler, and so incorrect use (passing anything but a `Tuple`-typed value) warrants an assert.